standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -2673,4 +2673,13 @@ private void getStatsTableListResult(
       query.closeAll();
     }
   }
+
+  public void lockDbTable(String tableName) throws MetaException {
+    String lockCommand = "lock table \"" + tableName + "\" in exclusive mode";
+    try {
+      executeNoResult(lockCommand);
+    } catch (SQLException sqle) {
```

Comment:
i think retry exception should be handled

Reply:
Done. Wrapped it in RetryingExecutor.